From 609abd505f52bf39fd86facbfc588881ff428f2c Mon Sep 17 00:00:00 2001
From: "Suresh, Prashanth" <prashanth.suresh@intel.com>
Date: Tue, 12 Jul 2022 09:33:22 +0530
Subject: [PATCH] Add config ini as part of package.

Add ini file as part of config and update paths.

Tracked-On:
Signed-off-by: Suresh, Prashanth <prashanth.suresh@intel.com>

diff --git a/scripts/setup_civ_ini.sh b/scripts/setup_civ_ini.sh
new file mode 100755
index 0000000..aab7697
--- /dev/null
+++ b/scripts/setup_civ_ini.sh
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+# Copyright (c) 2022 Intel Corporation.
+# All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -eE
+CIV_WORK_DIR=$(pwd)
+USER_HOME=$(getent passwd $SUDO_USER | cut -d: -f6)
+
+function setup_civ_ini() {
+        filename=$(ls caas-flashfiles*)
+        sed -i "/^\[global\]$/,/^\[/ s#^name.*#name=civ-sriov#" $USER_HOME/.intel/.civ/civ-sriov.ini
+        sed -i "/^\[global\]$/,/^\[/ s#^flashfiles.*#flashfiles=$CIV_WORK_DIR\/$filename#" $USER_HOME/.intel/.civ/civ-sriov.ini
+        sed -i "/^\[firmware\]$/,/^\[/ s#^path.*#path=$CIV_WORK_DIR\/OVMF.fd#" $USER_HOME/.intel/.civ/civ-sriov.ini
+        sed -i "/^\[disk\]$/,/^\[/ s#^path.*#path=$CIV_WORK_DIR\/android.qcow2#" $USER_HOME/.intel/.civ/civ-sriov.ini
+        sed -i "/^\[graphics\]$/,/^\[/ s#^type.*#type=SRIOV#" $USER_HOME/.intel/.civ/civ-sriov.ini
+
+        sed -i "/^\[graphics\]$/,/^\[/ s/^gvtg_version.*/#gvtg_version=/" $USER_HOME/.intel/.civ/civ-sriov.ini
+        sed -i "/^\[graphics\]$/,/^\[/ s/^vgpu_uuid.*/#vgpu_uuid=/" $USER_HOME/.intel/.civ/civ-sriov.ini
+
+        sed -i "/^\[net\]$/,/^\[/ s/^model.*/#model=/" $USER_HOME/.intel/.civ/civ-sriov.ini
+
+        sed -i "/^\[vtpm\]$/,/^\[/ s#^data_dir.*#data_dir=$CIV_WORK_DIR\/vtpm0#" $USER_HOME/.intel/.civ/civ-sriov.ini
+
+        sed -i "/^\[rpmb\]$/,/^\[/ s#^bin_path.*#bin_path=$CIV_WORK_DIR\/scripts\/rpmb_dev#" $USER_HOME/.intel/.civ/civ-sriov.ini
+        sed -i "/^\[rpmb\]$/,/^\[/ s#^data_dir.*#data_dir=$CIV_WORK_DIR#" $USER_HOME/.intel/.civ/civ-sriov.ini
+
+        sed -i "/^\[aaf\]$/,/^\[/ s#^path.*#path=$CIV_WORK_DIR/scripts/aaf#" $USER_HOME/.intel/.civ/civ-sriov.ini
+        sed -i "/^\[aaf\]$/,/^\[/ s#^support_suspend.*#support_suspend=disable#" $USER_HOME/.intel/.civ/civ-sriov.ini
+
+        sed -i "/^\[passthrough\]$/,/^\[/ s/^passthrough_pci/#passthrough_pci/" $USER_HOME/.intel/.civ/civ-sriov.ini
+}
+
+function copy_civ_ini() {
+        mkdir -p $USER_HOME/.intel/.civ/
+        cp $CIV_WORK_DIR/scripts/civ-1.ini $USER_HOME/.intel/.civ/civ-sriov.ini
+        chmod 0666 $USER_HOME/.intel/.civ/civ-sriov.ini
+}
+
+copy_civ_ini
+setup_civ_ini
+
diff --git a/scripts/setup_host.sh b/scripts/setup_host.sh
index ad696fa..3a29e0d 100755
--- a/scripts/setup_host.sh
+++ b/scripts/setup_host.sh
@@ -585,6 +585,10 @@ function parse_arg() {
     done
 }
 
+function setup_civ_ini() {
+        $CIV_WORK_DIR/scripts/setup_civ_ini.sh
+}
+
 function create_aaf_dir() {
         mkdir -p $CIV_WORK_DIR/scripts/aaf
 }
@@ -609,6 +613,7 @@ ubu_update_fw
 
 install_vm_manager
 create_aaf_dir
+setup_civ_ini
 
 prepare_required_scripts
 ubu_install_swtpm
-- 
2.39.1

